### PR TITLE
Add apicurio jsonserde support

### PR DESCRIPTION
**Pull request by carlesarnal**
This PR includes the initial support for json schema serialization using Apicurio Registry serdes. For now, autodiscovery of the channel and appropriate serde is not possible, so it must be manually configured.

### CODE REVIEW

**Comment by carlesarnal** [comment]
@cescoffier do you mind approving the CI run? I have the quickstart ready locally and the tests passing for native as well.

**Comment by gsmet** [comment]
@carlesarnal hey, thanks for this PR. Ideally, make sure CI passes in your fork before we approve the CI run here.

Also, probably a good idea to squash the commits to be a bit more semantic. Not sure if you need one big one or several ones but we don't need to see all the small iterations.

Thanks!

**Comment by carlesarnal** [comment]
> @carlesarnal hey, thanks for this PR. Ideally, make sure CI passes in your fork before we approve the CI run here.
> 
> Also, probably a good idea to squash the commits to be a bit more semantic. Not sure if you need one big one or several ones but we don't need to see all the small iterations.
> 
> Thanks!

Hey, yes, I realised now that the CI jobs are running on my fork, I'll make sure they pass there and squash the commits, tyvm.

**Comment by carlesarnal** [comment]
@gsmet all the tests are green on my fork except for the JVM 17 on Windows that, for some reason cannot find a valid docker env. Any hint on what might be happening?

**Comment by carlesarnal** [comment]
> @gsmet all the tests are green on my fork except for the JVM 17 on Windows that, for some reason cannot find a valid docker env. Any hint on what might be happening?

Ok, I might have figured out the reason, I was defaulting the test execution to true no matter the profile, I have changed the default values appropriately, let's see what CI has to say, sorry about the noise.

**Comment by carlesarnal** [comment]
@cescoffier at your convenience, CI on my fork is green, including native tests etc, so this is ready for your review. Thanks.

File: extensions/schema-registry/apicurio/json-schema/runtime/src/main/resources/META-INF/quarkus-extension.yaml
Comment (cescoffier) [inline extensions/schema-registry/apicurio/json-schema/runtime/src/main/resources/META-INF/quarkus-extension.yaml:7]: Should we add kafka?

**Review by cescoffier — APPROVED** [review]
LGTM

It would be great to:

1. implement serde discovery if possible (not totally sure it it possible) 
2. Make sure native is working as expected.

**Comment by carlesarnal** [comment]
> LGTM
> 
> It would be great to:
> 
> 1. implement serde discovery if possible (not totally sure it it possible)
> 2. Make sure native is working as expected.

1. It may be possible by making it depend on the schema location property since that property is only configurable for the json schema serializer. I would need some guidance on how to implement it.
2. Native is working fine so far, the tests have been included in the native tests json and are passing on my fork and I already have the quickstart ready [here](https://github.com/quarkusio/quarkus-quickstarts/pull/1361/files#diff-3683a52f06116bf78445468c56f0cd2f62f6ae96fa582176e8bb2473f37be425) and the native tests is passing there as well with no issues with a local build of Quarkus. The only requisite in this case is to add the json schema that will be used for serialization to the `quarkus.native.resources.includes` property, something like [this](https://github.com/quarkusio/quarkus-quickstarts/pull/1361/files#diff-95de300c5ee82f46f0e8984ccabc15b3a6cf6d30a66ce349b286af8d747e8929R3).

File: extensions/schema-registry/apicurio/json-schema/runtime/src/main/resources/META-INF/quarkus-extension.yaml
Comment (carlesarnal) [reply extensions/schema-registry/apicurio/json-schema/runtime/src/main/resources/META-INF/quarkus-extension.yaml:7]: Your call, but sounds reasonable to me. This is bound to the kafka use case.

**Review by alesj — APPROVED** [review]

**Pull request merged**
